src/core/logical_layers/utils/layersOrder/layersOrder.ts

```diff
@@ -126,37 +190,40 @@ export class LayersOrderManager {
     // return undefined so it wouldn't draw under the basemap
     if (!customLayers.length) return undefined;
 
-    // Create map { type: upper layer with this type }
-    const mountedBottomLayersByType = new Map<
+    const mountedOrderedLayers = new Map<
       LayersType,
-      maplibregl.AnyLayer
+      { layer: maplibregl.AnyLayer; category?: LayerCategory }[]
     >();
-    let n = 0;
-    // this loop will come from bottom to top layers
-    while (n < customLayers.length) {
-      const layer = customLayers[n];
-      // set the lowest layer for it's type
-      if (!mountedBottomLayersByType.has(layer.type)) {
-        mountedBottomLayersByType.set(layer.type, layer);
-      }
-      n++;
-    }
-    return mountedBottomLayersByType;
+
+    // this loop will come from bottom to top layers (from backgrounds and rasters to lines and symbols)
+    customLayers.forEach((layer) => {
```

Comment:
There was a reason why I used here while
https://www.measurethat.net/Benchmarks/ShowResult/372430
https://www.measurethat.net/Benchmarks/Show/9739/0/array-loop-vs-foreach-vs-map-vs-while#latest_results_block

Reply:
This cycle works with pretty big array and performance here very important

Reply:
chromium, on the other hand, is the opposite
https://www.measurethat.net/Benchmarks/Show/9739/0/array-loop-vs-foreach-vs-map-vs-while#latest_results_block

Since the slowest version of firefox is still many times faster than the fastest in chromium, and it is forEach, it's ok to use forEach